buildkite/resource_pipeline_team.go

```diff
@@ -0,0 +1,232 @@
+		return
+	}
+
+	apiResponse, err := getNode(
+		tp.client.genqlient,
+		state.Id.ValueString(),
+	)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to read team pipeline",
+			fmt.Sprintf("Unable to read team pipeline: %s", err.Error()),
+		)
+	}
+
+	// Convert from Node to getNodeNodeTeamPipeline type
+	if teamPipelineNode, ok := apiResponse.GetNode().(*getNodeNodeTeamPipeline); ok {
+		if teamPipelineNode == nil {
+			resp.Diagnostics.AddError(
+				"Unable to get team pipeline",
+				"Error getting team pipeline: nil response",
+			)
+			return
+		}
+		updateTeamPipelineResourceState(&state, *teamPipelineNode)
+		resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+	} else {
+		// Resource not found, remove from state
+		resp.Diagnostics.AddWarning("Team pipeline resource not found", "Removing team pipeline from state")
+		resp.State.RemoveResource(ctx)
+	}
+
+}
+
+func (tp *pipelineTeamResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func (tp *pipelineTeamResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var state pipelineTeamResourceModel
+	var accessLevel string
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	resp.Diagnostics.Append(req.Plan.GetAttribute(ctx, path.Root("access_level"), &accessLevel)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	_, err := updateTeamPipeline(tp.client.genqlient,
+		state.Id.ValueString(),
+		PipelineAccessLevels(accessLevel),
+	)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to update team pipeline",
+			fmt.Sprintf("Unable to update team pipeline: %s", err.Error()),
+		)
+		return
+	}
+
+	// Update state with values from API response/plan
+	state.AccessLevel = types.StringValue(accessLevel)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (tp *pipelineTeamResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state pipelineTeamResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	_, err := deleteTeamPipeline(tp.client.genqlient, state.Id.ValueString())
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to delete team pipeline",
+			fmt.Sprintf("Unable to delete team pipeline: %s", err.Error()),
+		)
+		return
+	}
+}
+
+func updateTeamPipelineResourceState(tpState *pipelineTeamResourceModel, tpNode getNodeNodeTeamPipeline) {
+	tpState.Id = types.StringValue(tpNode.Id)
+	tpState.Uuid = types.StringValue(tpNode.Uuid)
+	tpState.TeamId = types.StringValue(tpNode.Team.Id)
+	tpState.PipelineId = types.StringValue(tpNode.Pipeline.Id)
+	tpState.AccessLevel = types.StringValue(string(tpNode.PipelineAccessLevel))
+}
+
+func findTeamInPipelineNode(client *graphql.Client, tpState *pipelineTeamResourceModel) bool {
```

Comment:
This looks unused now. Can it be removed?

Reply:
Same as the issue in line 74 comment, the unused function got reverted in this commit [Fix: Missing } on setPipelineModel and removed unused function blocks](https://github.com/buildkite/terraform-provider-buildkite/pull/351/commits/4d9cd24cff75d2785f6796d07ce666600f2fc52e). I will make sure it got reverted to that.